### PR TITLE
[PAN-3200] migrate privacy storage

### PR DESCRIPTION
**Pull request by iikirilov**

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/MessageFrame.java [hunk 1]
```diff
@@ -871,6 +877,15 @@ public Boolean isPersistingState() {
     return isPersistingState;
   }
 
+  /**
+   * Returns whether Message calls will be persisted
+   *
+   * @return whether Message calls will be persisted
+   */
+  public Hash getTransactionHash() {
```
Comment (pinges) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/MessageFrame.java]: This method is used to build the PrivateBlockMetadata, but I don't fully understand what the transaction hash is needed for ...

Comment (iikirilov) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/MessageFrame.java]: this is the PMT transaction hash

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/PrivateStateKeyValueStorage.java [hunk 2]
```diff
@@ -99,9 +115,125 @@ public boolean isWorldStateAvailable(final Bytes32 rootHash) {
...
   }
 
-  @Override
-  public PrivateStateStorage.Updater updater() {
-    return new PrivateStateKeyValueStorage.Updater(keyValueStorage.startTransaction());
+  private PrivateBlockMetadata rlpDecodePrivateBlockMetadata(final BytesValue bytes) {
+    return PrivateBlockMetadata.readFrom(RLP.input(bytes));
+  }
+
+  private void performStateRootMigration(
```
[the remark this answers is not in the document]
Comment (iikirilov) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/privacy/storage/PrivateStateKeyValueStorage.java]: yes

**Comment by MadelineMurray** [comment]
Superseeded by Lucas's PR

**Pull request closed without merging**